### PR TITLE
Add texture header file unpack command to flutter_tools for GLFW shell (#30718).

### LINKED ISSUE #30718: Support Texture widget in GLFW shell
See google/flutter-desktop-embedding#107 for discussion

### PR DESCRIPTION
**Pull request by cloudwebrtc**
## Description

Fix #30718

Depends on the following:
https://github.com/flutter/engine/pull/9822
https://github.com/flutter/buildroot/pull/275

## Related Issues

https://github.com/flutter/flutter/issues/30718

## Tests
I did the tests on Windows 10 x64 and Ubuntu Linux 16.04 x64. 
Correctly copy flutter_texture_registrar.h from src/out/host_debug and complete the flutter app compilation.
Use the following command:
`flutter build linux --local-engine=host_debug --debug`
`flutter build windows --local-engine=host_debug --debug`

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by zanderso** [comment]
Marking work in progress while waiting for the engine PRs to land.

File: packages/flutter_tools/lib/src/build_system/targets/windows.dart [hunk 1]
```diff
@@ -41,6 +41,7 @@ const Target unpackWindows = Target(
     Source.pattern('{PROJECT_DIR}/windows/flutter/flutter_export.h'),
     Source.pattern('{PROJECT_DIR}/windows/flutter/flutter_messenger.h'),
     Source.pattern('{PROJECT_DIR}/windows/flutter/flutter_plugin_registrar.h'),
+    Source.pattern('{PROJECT_DIR}/linux/flutter/flutter_texture_registrar.h'),
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/build_system/targets/windows.dart]: linux->windows

**Comment by zanderso** [comment]
@cloudwebrtc what is the status of this PR?

**Comment by jonahwilliams** [comment]
Because of the lack of activity on this PR I'm going to close it to remove from our review queue. Please feel free to send us a new PR or discuss in an issue anytime.

**Pull request closed without merging**